hayagriva.schema.json

```diff
@@ -0,0 +1,436 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://raw.githubusercontent.com/typst/hayagriva/main/hayagriva.schema.json",
+    "title": "Hayagriva Bibliography Format",
+    "description": "A bibliography management format for the modern age.",
+    "type": "object",
+    "additionalProperties": false,
+    "patternProperties": {
+        "([A-Z]|[a-z]|[0-9]|)+": {
+            "$dynamicAnchor": "entry",
+            "$comment": "The key for a reference item can be series of alphanumeric characters that are uninterrupted by whitespace.",
+            "type": "object",
+            "description": "An entry in the bibliography. The citation key for the entry can be any alphanumeric character.",
+            "examples": [
+                {
+                    "plaque": {
+                        "type": "Misc",
+                        "title": "Informational plaque about Jacoby's 1967 photos",
+                        "publisher": "Stiftung Reinbeckhallen",
+                        "location": "Berlin, Germany",
+                        "date": 2020,
+                        "parent": {
+                            "type": "Artwork",
+                            "date": 1967,
+                            "author": "Jacoby, Max",
+                            "parent": {
+                                "type": "Anthology",
+                                "title": "Bleibtreustraße",
+                                "archive": "Landesmuseum Koblenz",
+                                "archive-location": "Koblenz, Germany"
+                            }
+                        }
+                    }
+                }
+            ],
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "examples": [
+                        "video"
+                    ],
+                    "enum": [
+                        "article",
+                        "chapter",
+                        "entry",
+                        "anthos",
+                        "report",
+                        "thesis",
+                        "web",
+                        "scene",
+                        "artwork",
+                        "patent",
+                        "case",
+                        "newspaper",
+                        "legislation",
+                        "manuscript",
+                        "original",
+                        "post",
+                        "misc",
+                        "performance",
+                        "periodical",
+                        "proceedings",
+                        "book",
+                        "blog",
+                        "reference",
+                        "conference",
+                        "anthology",
+                        "repository",
+                        "thread",
+                        "video",
+                        "audio",
+                        "exhibition"
```

Comment:
Types are case-insensitive. In particular, they can start with an uppercase letter. I don't think that there is a particularly clean solution to encoding this right now though: https://stackoverflow.com/questions/26604730/can-json-schema-enums-be-case-insensitive